python/lib/cloudutils/serviceConfig.py

```diff
@@ -495,6 +495,27 @@ def config(self):
             logging.debug(formatExceptionInfo())
             return False
 
+class hostConfig(serviceCfgBase):
+    def __init__(self, syscfg):
+        super(hostConfig, self).__init__(syscfg)
+        self.serviceName = "Host"
+
+    def config(self):
+        try:
+            cfo = configFileOps("/etc/cloudstack/agent/agent.properties", self)
+            reservedMemory = float(cfo.getEntry("host.reserved.mem.mb").strip() or 1024)
+            totalMemory = float(bash("awk '/MemTotal/ { printf \"%.3f \\n\", $2/1024 }' /proc/meminfo").getStdout().strip())
+            if totalMemory < reservedMemory :
+                raise CloudRuntimeException("CloudStack requires more than %d MB memory since %d MB is reserved" %(reservedMemory, reservedMemory))
+
+            if int(bash('ip a | grep "^\\w" | grep -iv "^lo" | wc -l').getStdout()) < 1 :
```

Comment:
It may fail for env where hosts are connected on wifi? And will this command work on all distros?

Reply:
It will work since it checks for any nic. No explicit regex such as starts with 'e', etc. IP is present by default on all supported distros

Reply:
So just to confirm it counts any nics and doesn't ignore any nics, we shouldn't assume nics name will start with "e" though.
You can instead change the logic to count physical nics, for ex use something from:
lshw -class network
Or,
find /sys/class/net -type l -not -lname '*virtual*' -printf '%f\n'

Reply:
Yes, it counts all nics excluding loopback
In some minimal instances lshw is not present, so using IP -a since that is pretty much always there